yt_dlp/extractor/cbc.py

```diff
@@ -209,289 +185,243 @@ def _real_extract(self, url):
+                    'duration': episode.get('duration'),
+                    'categories': [episode.get('category')],
+                })
 
-class CBCOlympicsIE(InfoExtractor):
-    IE_NAME = 'cbc.ca:olympics'
-    _VALID_URL = r'https?://olympics\.cbc\.ca/video/[^/]+/(?P<id>[^/?#]+)'
-    _TESTS = [{
-        'url': 'https://olympics.cbc.ca/video/whats-on-tv/olympic-morning-featuring-the-opening-ceremony/',
-        'only_matching': True,
-    }]
+            thumbnail = None
+            tn_uri = season_info.get('image')
+            # the-national was observed to use a "data:image/png;base64"
+            # URI for their 'image' value. The image was 1x1, and is
+            # probably just a placeholder, so it is ignored.
+            if tn_uri is not None and not tn_uri.startswith('data:'):
+                thumbnail = tn_uri
+
+            return {
+                '_type': 'playlist',
+                'entries': episodes,
+                'id': season_id,
+                'title': season_info['title'],
+                'description': season_info.get('description'),
+                'thumbnail': thumbnail,
+                'series': show_info.get('title'),
+                'season_number': season_info.get('season'),
+                'season': season_info['title'],
+            }
+
+        elif re.fullmatch(self._VIDEO_ID, url_id) is None:
+            # Not a playlist or video
+            raise ExtractorError(f"Could't recognize ID '{url_id}' as a playlist or video")
+
+        # It's a single video
+
+        video_id = url_id
+        video_info = self._download_json(self._API_VIDEO + video_id, video_id)
+        m3u8_info = self._download_json(video_info['playSession']['url'], video_id)
+
+        if m3u8_info.get('errorCode') == 1:
+            # errorCode 1 means geo-blocked
+            self.raise_geo_restricted(countries=['CA'])
+        elif m3u8_info.get('errorCode') not in [35, 0, None]:
+            # 35 means media unavailable, which is handled below. 0 means no
+            # error. None is there just in case. Any other errorCode can't be
+            # handled as we don't know what it is.
+            raise ExtractorError(
+                f"Unknown errorCode {m3u8_info.get('errorCode')}, with message: {m3u8_info.get('message')}"
+            )
+
+        m3u8_url = m3u8_info.get('url')
+
+        # Sometimes the m3u8 URL is a not available, and errorCode is 35.
+        # This might only happen with member content. But just retrying seems
+        # to work fine.
+        i = 0
+        while m3u8_url is None:
+            if i == 15:
+                # Didn't work after 15 tries, give up
+                raise ExtractorError("Couldn't retrieve m3u8 URL")
+            m3u8_url = self._download_json(
+                video_info['playSession']['url'], video_id).get('url')
+            i += 1
+
+        formats = self._extract_m3u8_formats_and_subtitles(
+            m3u8_url, video_id, m3u8_id='hls')[0]
```

Comment:
if there are no subs, use `_extract_m3u8_formats`. If there are, dont discard it